src/components/deploymentConfig/DeploymentConfig.tsx

```diff
@@ -438,13 +479,64 @@ export default function DeploymentConfig({
 
         if (state.chartConfig.id) {
             //update flow, might have overridden
+            handleConfirmationDialog(true)
+        } else {
+            //create flow
+            checkForLockedChanges()
+
+        }
+    }
+
+    const checkForLockedChanges = async () => {
+        dispatch({
+            type: DeploymentConfigStateActionTypes.loading,
+            payload: true,
+        })
+        try {
+            const requestBody = prepareDataToSave(true)
+            let res, api
+            if (isProtected) {
+                res = await checkForProtectedLockedChanges()
+            } else {
+                api=getIfLockedConfigNonProtected
+                res = await api(requestBody, baseDeploymentAbortController.signal)
+            }
+            if (res.result.isLockConfigError ) {
+                setLockedOverride(res.result?.lockedOverride)
+                openLockedDiffDrawer()
+                return
+            } else if (isProtected) {
+                toggleSaveChangesModal()
+                return
+            }
+            if (state.chartConfig.id) {
```

Comment:
We can add this in elseif and remove the return

Reply:
done